### PR TITLE
Publish course

### PR DESCRIPTION
**Pull request by MartinBentley**
### Context
These are the changes to  allow publishing of a course to search and compare
### Changes proposed in this pull request
The course mapper has been moved into the Api and an endpoint created to return as search and compare course for preview.
Also there is a change in the Publish process to call a new endpoint on the Api to publish a course.

### CODE REVIEW

File: src/ui/Controllers/CourseController.cs [hunk 1]
```diff
@@ -74,9 +72,11 @@ public async Task<IActionResult> ShowPublish(string instCode, string accrediting
                 return await Show(instCode, accreditingProviderId, ucasCode);
             }
 
-            var result = await _manageApi.PublishEnrichmentCourse(instCode, ucasCode);
+            var statusResult = await _manageApi.PublishEnrichmentCourse(instCode, ucasCode);
+            //dont do this yet
+            //var publishResult = await _manageApi.PublishCourse(instCode, ucasCode);
```
Comment (defong) [inline src/ui/Controllers/CourseController.cs]: Is there any need to separate this out?

File: src/ui/ManageApi.cs [hunk 2]
```diff
@@ -87,12 +94,33 @@ public async Task SaveEnrichmentCourse(string instCode, string ucasCode, CourseE
...
         }
+
+        public async Task<bool> PublishCourse(string instCode, string courseCode)
+        {
+            var result = await _apiClient.Publish_PublishAsync(instCode, courseCode);
+
+            return result;
+        }
+
+        private SearchAndCompare.Domain.Models.Course ConvertCourse(ApiClient.Course2 course)//TODO sort this Course2 its pants
```
Comment (defong) [inline src/ui/ManageApi.cs]: Yes its pants

File: src/ui/ManageApi.cs [hunk 3]
```diff
@@ -87,12 +90,22 @@ public async Task SaveEnrichmentCourse(string instCode, string ucasCode, CourseE
...
             await _apiClient.Enrichment_SaveCourseAsync(instCode, ucasCode, course);
         }
-        
-        public async Task<bool> PublishEnrichmentCourse(string ucasCode, string courseCode)
+
+        public async Task<SearchAndCompare.Domain.Models.Course> GetSearchAndCompareCourse(string ucasCode, string courseCode)
         {
-            await _apiClient.Enrichment_PublishCourseAsync(ucasCode, courseCode);
+            var result = await _apiClient.Publish_GetSearchAndCompareCourseAsync(ucasCode, courseCode);
+            var jsonCourse = JsonConvert.SerializeObject(result, _jsonSerializerSettings);
```
Comment (the-daniel-rothig) [inline src/ui/ManageApi.cs]: It would be good to have a comment here explaining why this is happening

Comment (MartinBentley) [reply src/ui/ManageApi.cs]: Added comments

**Pull request merged**